services/addons/images/iss_health_check/iss_health_checker.py

```diff
@@ -1,118 +1,124 @@
-        "INFO" if "LOGGING_LEVEL" not in os.environ else os.environ["LOGGING_LEVEL"]
-    )
-    logging.basicConfig(format="%(levelname)s:%(message)s", level=log_level)
-
-    scms_statuses = get_scms_status_data()
-    insert_scms_data(scms_statuses)
+from datetime import datetime
+import requests
+import logging
+import os
+import iss_token
+import common.pgquery as pgquery
+
+
+def get_rsu_data():
+    result = {}
+    query = (
+        "SELECT jsonb_build_object('rsu_id', rsu_id, 'iss_scms_id', iss_scms_id) "
+        "FROM public.rsus "
+        "WHERE iss_scms_id IS NOT NULL "
+        "ORDER BY rsu_id"
+    )
+    data = pgquery.query_db(query)
+
+    logging.debug("Parsing results...")
+    for point in data:
+        point_dict = dict(point[0])
+        result[point_dict["iss_scms_id"]] = {"rsu_id": point_dict["rsu_id"]}
+
+    return result
+
+
+def get_scms_status_data():
+    rsu_data = get_rsu_data()
+
+    # Create GET request headers
+    iss_headers = {}
+    iss_headers["x-api-key"] = iss_token.get_token()
+
+    # Create the GET request string
+    iss_base = os.environ["ISS_SCMS_VEHICLE_REST_ENDPOINT"]
+    project_id = os.environ["ISS_PROJECT_ID"]
+    page_size = 200
+    page = 0
+    messages_processed = 0
+
+    # Loop through all pages of enrolled devices
+    while True:
+        iss_request = iss_base + "?pageSize={}&page={}&project_id={}".format(
+            page_size, page, project_id
+        )
+        logging.debug("GET: " + iss_request)
+        response = requests.get(iss_request, headers=iss_headers)
+        enrollment_list = response.json()["data"]
+
+        if len(enrollment_list) == 0:
+            break
+
+        # Loop through each device on current page
+        for enrollment_status in enrollment_list:
+            if enrollment_status["_id"] in rsu_data:
+                rsu_data[enrollment_status["_id"]][
+                    "provisionerCompany"
+                ] = enrollment_status["provisionerCompany_id"]
+                rsu_data[enrollment_status["_id"]]["entityType"] = enrollment_status[
+                    "entityType"
+                ]
+                rsu_data[enrollment_status["_id"]]["project_id"] = enrollment_status[
+                    "project_id"
+                ]
+                rsu_data[enrollment_status["_id"]]["deviceHealth"] = enrollment_status[
+                    "deviceHealth"
+                ]
+
+                # If the device has yet to download its first set of certs, set the expiration time to when it was enrolled
+                if "authorizationCertInfo" in enrollment_status["enrollments"][0]:
+                    rsu_data[enrollment_status["_id"]][
+                        "expiration"
+                    ] = enrollment_status["enrollments"][0]["authorizationCertInfo"][
+                        "expireTimeOfLatestDownloadedCert"
+                    ]
+                else:
+                    rsu_data[enrollment_status["_id"]]["expiration"] = None

```

Comment:
A suggestion of using data structure like @dataclass or NamedTuple to define the rsu_data so that it is clear what attributes we expect from the rsu_data.

Reply:
A wrapper class for rsu_data has been created in `iss_health_checker.py` to improve clarity regarding what attributes we expect to be setting in the dictionary.